### PR TITLE
Implementing CSV ingestion endpoint

### LINKED ISSUE #131: Convert CSV data (from PO) to JSON for Ingest
~**This story is blocked until Brent delivers real data in January**~
We will work with dummy data 

To prepare for PO demo at end of January....

Brent is to provide csvs of "real data" likely of size 1,000-10,000 records in January. These records will have a key of lei and attribute of nace codes.

Write a script (python?) to convert this csv data to json so we can ingest them into our application. 

## Acceptance Criteria 
1. Add ADR to define the possible ways that we could do this (new endpoint, typescript converting the csv, standalone tool that converst csv to json)
2. Add original csvs, json files and script to [demo data folder](https://github.com/microsoft/ccf-app-samples/tree/main/data-reconciliation-app/demo/data)

### PR DESCRIPTION
**Pull request by julioalex-rezende**
closes #131 
Implementing CSV Ingestion endpoint.
- `papaparse` added to devDependencies
- new endpoint created: `/app/csc/ingest`
- `ingest-csv.ts` file created to handle CSV endpoint
- `test.sh` script slightly changed so now member0 submits their data through the CSV endpoint.
- some tweaks in the ADR to document CSV ingestion and default schema

CSV is now accepted as input file through `/app/csv/ingest` endpoint
As the focus of this is just a sample, no concerns were taken with respect to performance or data size.

Once in the application, CSV data is read from the http request body as text and converted to JSON using `papaparse`, which handles different validations to the file under the hoods.

Once converted to JSON, the code takes the same flow as if JSON was used directly.

### CODE REVIEW

File: data-reconciliation-app/docs/adr/01-data-ingest.md [hunk 1]
```diff
@@ -15,15 +15,17 @@ Gathered requirements:
...
 ## Decision
 
 ## API
 
-### Endpoint
+### JSON Ingestion Endpoint
 
 - Description
-  - Ingest member or user data into our application and add data to K-V store
+  - Ingest member or user data into our application and add data to K-V store via JSON
```
Comment (ross-p-smith) [inline data-reconciliation-app/docs/adr/01-data-ingest.md:28]:
```suggestion
  - Ingest member or user data and add to K-V store via JSON
```

File: data-reconciliation-app/docs/adr/01-data-ingest.md [hunk 2]
```diff
@@ -15,15 +15,17 @@ Gathered requirements:
   - one unique identifier (string)
   - one attribute associated with the unique identifier (string)
 - There is likely going to be a UI on top of our ingest API...so we will treat our app like a backend system. Therefore, we will accept JSON into our ingest API.
+- As captured in [Ingestion Format ADR](./07-ingestion-format.md), Many financial institutions transmit their data in Comma Seperated Value (CSV) files. Therefore we will also accept CSV for data ingestion, which will then be converted and treated as a JSON object in the application.
```
Comment (ross-p-smith) [inline data-reconciliation-app/docs/adr/01-data-ingest.md:18]:
```suggestion
- As captured in [Ingestion Format ADR](./07-ingestion-format.md), many financial institutions transmit their data in Comma Separated Value (CSV) files. Therefore we will also accept CSV for data ingestion.
```

File: data-reconciliation-app/docs/adr/01-data-ingest.md [hunk 3]
```diff
@@ -44,6 +46,31 @@ Gathered requirements:
     - Status: 400
     - Status Text: "Validation Error"
 
+### CSV Ingestion Endpoint
+
+- Description
+  - Ingest member or user data into our application and add data to K-V store via CSV.
```
Comment (ross-p-smith) [inline data-reconciliation-app/docs/adr/01-data-ingest.md:52]:
```suggestion
  - Ingest member or user data and add to K-V store via CSV.
```

File: data-reconciliation-app/src/endpoints/all.ts [hunk 4]
```diff
@@ -1,2 +1,3 @@
 export * from "./ingest";
```
Comment (ross-p-smith) [inline data-reconciliation-app/src/endpoints/all.ts:1]: nit: do we want to change the original file to ingest-json.ts?!?!

**Review by ross-p-smith — APPROVED** [review]
LGTM!

**Review by Aymalla — APPROVED** [review]
LGTM

File: data-reconciliation-app/src/endpoints/all.ts (hunk 4 above)
Comment (julioalex-rezende) [reply data-reconciliation-app/src/endpoints/all.ts:1]: IMHO we can keep it as it is given that it's our default model, and we are providing an extra feature to allow CSV.

Dropping this for now, we can revisit it if you disagree :)

**Pull request merged**